### PR TITLE
Add support to metrics framework to auto-generate documentation for metrics

### PR DESCRIPTION
**Pull request by logicalhan**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR introduces the ability to auto-generate documentation for metrics used in Kubernetes.

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3498
```

### CODE REVIEW

File: test/instrumentation/testdata/documentation.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+title: Kubernetes Metrics
+content_type: instrumentation
+---
+
+
+## Metrics
+
+These are the metrics which are exported in Kubernetes components (i.e. kube-apiserver, scheduler, kube-controller-manager, kube-proxy, cloud-controller-manager). 
+
+(auto-generated 2022 Oct 25)
```
Comment (sftim) [inline test/instrumentation/testdata/documentation.md:11]: If this file isn't automatically kept current / tested for staleness, should we commit it?

Comment (logicalhan) [reply test/instrumentation/testdata/documentation.md:11]: My plan was to run this each code freeze.

Comment (sftim) [reply test/instrumentation/testdata/documentation.md:11]: I'm wary of this.

>  If a piece of information doesn't have to be correct for the system to work, sooner or later it won't be.

- https://utcc.utoronto.ca/~cks/space/blog/sysadmin/SysadminAphorismII

Comment (logicalhan) [reply test/instrumentation/testdata/documentation.md:11]: I fail to see how omitting this is better than what I am proposing.

Comment (sftim) [reply test/instrumentation/testdata/documentation.md]: What I'd do: signpost readers to the document for the latest release (which will form part of the docs on the website). If anyone wants to know how that's different from what's on the _master_ branch, instead signpost them to the script that lets them find out the very latest page.

I really have seen a lot of stale docs in my career, and they nearly all started out with great intentions to keep them current.

Comment (sftim) [reply test/instrumentation/testdata/documentation.md]: Website docs PR: https://github.com/kubernetes/website/pull/37522

Comment (logicalhan) [reply test/instrumentation/testdata/documentation.md]: I am happy to abandon my pursuit of documenting metrics if the process to do so exceeds my patience to pursue this.

Comment (sftim) [reply test/instrumentation/testdata/documentation.md]: (it's OK to resolve this feedback)

**Pull request merged**